pkg/blockbuilder/kafkautil.go

```diff
@@ -60,6 +60,11 @@ func getGroupLag(ctx context.Context, admClient *kadm.Client, topic, group strin
 			if !ok {
 				return nil, fmt.Errorf("partition %d not found in fallback offsets for topic %s", partition, topic)
 			}
+			if o.Offset < startOffset.At {
+				// Skip the resolved fallback offset if it's before the partition's start offset (i.e. before the earliest offset of the partition).
+				// This should not happen in Kafka, but can happen in Kafka-compatible systems, e.g. Warpstream.
+				continue
```

Comment:
so here we continue which effectively skips this partition. Is the assumption that the fallbackOffset call will fix itself eventually? What if we've set fallbackOffset == 12h and warpstream retention == 12h: doesn't this make this race condition more likely to occur? 

Why not just use the start offset in this case?

Reply:
We pass all three sets of offsets to `kadm.CalculateGroupLagWithStartOffsets`. It uses the start offset if the partition doesn't have a commit, i.e. when the partition isn't in the `offsets`. So we should be good here.